platform/lib/platform_web/controllers/api/api_v2_controller.ex

```diff
@@ -92,6 +92,61 @@ defmodule PlatformWeb.APIV2Controller do
     end
   end
 
+  def create_media(conn, params) do
+    project_id = conn.assigns.token.project_id
+    project = Projects.get_project!(project_id)
+
+    cond do
+      not Permissions.can_api_token_create_media?(conn.assigns.token) ->
+        json(conn |> put_status(401), %{error: "unauthorized"})
+
+      true ->
+        # Generate attribute parameters for all attributes in the input
+        media_params =
+          params
+          |> Enum.reduce(%{}, fn {key, value}, acc ->
+            case Attribute.get_attribute(key, project: project) do
+              nil ->
+                acc
+
+              attr ->
+                # Merge the generated attribute change params
+                Map.merge(
+                  acc,
+                  Material.generate_attribute_change_params(
+                    attr,
+                    value,
+                    project
+                  )
+                )
+            end
+          end)
+          |> Map.put("project_id", project_id)
+
+        # We expect a JSON array of URLs in the incident creation flow
+        if not is_nil(params["urls"]) do
+          media_params |> Map.put("urls", Jason.encode!(params["urls"]))
```

Comment:
to catch silly user errors, perhaps we should verify all these start with `http://` or `https://`? I'm thinking about someone accidentally passing a lot of `example.com` URLs without a protocol and then wondering what's wrong.

Reply:
I know we don't provide any validation of URLs, but we could add _some_ validation to prevent obvious gotchas.